info.go

```diff
@@ -289,6 +289,14 @@ func newProgramInfoFromFd(fd *sys.FD) (*ProgramInfo, error) {
 		jitedSize:            info.JitedProgLen,
 		loadTime:             time.Duration(info.LoadTime),
 		verifiedInstructions: info.VerifiedInsns,
+		memlock:              0,
+	}
+
+	// Supplement OBJ_INFO with data from /proc/self/fdinfo. It contains fields
+	// like memlock that is not present in OBJ_INFO.
+	err2 := newProgramInfoFromProc(fd, &pi)
```

Comment:
As in https://github.com/cilium/ebpf/commit/21be59580a51a37397d2041c46d33c7c529b35b8, the reason for err1 and err2 existing is to be able to error if both approaches fail, but you added no such check at the bottom of the function. The result of `makeSecondCall` should go into `err1`.